tests/test_repositories.py

```diff
@@ -16,10 +16,94 @@ def test_round_trip(self, john, repo):
         expected = [john]
         assert list(observed) == expected
 
+        repo.remove(john)
+        observed = repo.get()
+        assert list(observed) == []
```

Comment:
It should error. I plan to implement this but thought it would be better to get feedback early one to see if im heading right direction with the implementation. I can of course add it to this PR if you like or i could create a seperate issue and PR. The choice is yours!

Reply:
OK. Happy to wait.